File: daily/Arrays/1317.Convert-integer-to-the-sum-of-two-no-zero-integers.go
package daily

func getNoZeroIntegers(n int) []int {
	for i := 1; i < n; i++ {
		a := i
		b := n - i
		if !hasZero(a) && !hasZero(b) {
			return []int{a, b}
		}
	}
	return nil
}

func hasZero(n int) bool {
	for n > 0 {
		if n%10 == 0 {
			return true
		}
		n /= 10
	}
	return false
}

File: daily/Arrays/1317.Convert-integer-to-the-sum-of-two-no-zero-integers.py
from typing import List


class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        for a in range(1, n):
            b = n - a
            if not self._has_zero(a) and not self._has_zero(b):
                return [a, b]
        return []

    def _has_zero(self, x: int) -> bool:
        while x > 0:
            if x % 10 == 0:
                return True
            x //= 10
        return False

